app/server/logging/jsondb/src/test/java/io/syndesis/server/logging/jsondb/controller/KubernetesSupportIT.java

```diff
@@ -0,0 +1,157 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.server.logging.jsondb.controller;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.sun.net.httpserver.HttpServer;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.PodList;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
+import io.syndesis.server.jsondb.JsonDB;
+import io.syndesis.server.openshift.OpenShiftService;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.skife.jdbi.v2.DBI;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class KubernetesSupportIT {
+
+    final AtomicInteger count = new AtomicInteger(0);
+
+    final AtomicBoolean running = new AtomicBoolean(false);
+
+    HttpServer server;
+
+    @Before
+    public void setup() throws IOException {
+        server = HttpServer.create(new InetSocketAddress("0.0.0.0", 0), 0);
+        server.setExecutor(Executors.newCachedThreadPool());
+        server.start();
+
+        running.set(true);
+        server.createContext("/api/v1/namespaces/syndesis/pods/pod/log", exchange -> {
+            exchange.sendResponseHeaders(200, 0);
+            try (PrintStream body = new PrintStream(exchange.getResponseBody())) {
+                while (running.get()) {
+                    int current = count.incrementAndGet();
+                    body.println(current);
+                    body.flush();
+
+                    if (current > 5) {
+                        // provoke a timeout after 5 lines sent, with the sleep
+                        // below it waits for 1.6 sec
+                        Thread.sleep(600);
+                    }
+                    Thread.sleep(1000);
+                }
+            } catch (final InterruptedException ignored) {
+                return;
+            } finally {
+                exchange.close();
+            }
+        });
+    }
+
+    @Test
+    public void shouldTolerateNetworkTimeouts() throws IOException, InterruptedException, ExecutionException {
```

Comment:
Any feeling how much extra time this tests add to the overall run ? looks like +5 sec or so ?

Reply:
About 30sec, it won't be run on CI as we don't have failsafe configured in this module though.